src/main/java/com/cloudbees/jenkins/support/impl/BrowserErrorCollector.java

```diff
@@ -0,0 +1,72 @@
+package com.cloudbees.jenkins.support.impl;
+
+import hudson.Extension;
+import hudson.model.RootAction;
+import org.apache.commons.io.IOUtils;
+import org.kohsuke.stapler.Stapler;
+import org.kohsuke.stapler.StaplerProxy;
+import org.kohsuke.stapler.StaplerRequest;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+/**
+ * A simple endpoint to collect unhandled browser JS errors.
+ *
+ * To make use of this, install a window.onerror handler,
+ * using a script tag similar to the following in client side javascript code:
+ *  <code>
+ *
+ *            window.onerror = function (msg, url, _lineNo, _columnNo, error) {
+ *                var http = new XMLHttpRequest();
+ *                http.open('POST', "/errorCollector", true);
+ *                http.send(msg + '\n' + url + '\n' + error.stack);
+ *                return false;
+ *            }
+ *
+ *
+ *  </code>
+ *
+ *         This will attempt to log the error with a maybe helpful JS stacktrace.
+ *
+ */
+@Extension
+public class BrowserErrorCollector implements RootAction, StaplerProxy {
```

Comment:
good idea re invisible. 

The reason for protected is unprotected means people can just throw anything in the log, which may be opening a slight can of worms (if it was logged separately to others, sure). Obviously log ration etc would need to protect. 

In the case of anon read only access, yes would be nice to have this, but at a slight increased risk.